kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/data/vector/DefaultSubFieldVector.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.defaults.internal.data.vector;
+
+import java.math.BigDecimal;
+import java.util.function.Function;
+import static java.util.Objects.requireNonNull;
+
+import io.delta.kernel.data.ArrayValue;
+import io.delta.kernel.data.ColumnVector;
+import io.delta.kernel.data.MapValue;
+import io.delta.kernel.data.Row;
+import io.delta.kernel.types.DataType;
+import io.delta.kernel.types.StructField;
+import io.delta.kernel.types.StructType;
+
+import static io.delta.kernel.defaults.internal.DefaultKernelUtils.checkArgument;
+
+/**
+ * {@link ColumnVector} wrapper on top of {@link Row} objects. This wrapper allows referencing
+ * any nested level column vector from a set of rows.
+ * TODO: We should change the {@link io.delta.kernel.defaults.client.DefaultJsonHandler} to
```

Comment:
is this TODO still needed?

Reply:
Moving the todo to a more appropriate spot